---
title: "assigning ASVs with DADA2"
output: html_document
date: "r Sys.Date()"
---



```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE,
                      fig.align = "center",
                      fig.path = "../figures/01_DADA2")
```

#symbolically link our data files

We need to have access to our Fastq files for out project

#Goals of this file:

1. Load in the raw sequencing data
2. Visualize and inspect sequencing data to assess the quality of the sequencing reads.
3. Filter by the quality and trim primers from sequencing data
4. Write new fastq files that include high quality and trimmed sequences 
5. Re-inspect the quality of the reads (intuition check to make sure that we have it)
6. Infer the error rate on the forward R1 and reverse R2 reads.
**DADA2 building the error model**
7. Apply the error model and infer ASVs on the forward and reverse reads individually.
8. Merge forward and reverse ASVs into "contiguous ASVs".
9. Generate an ASV count table ('otu_table' for phyloseq.)


output that we need:
1. ASV count table: 'otu_table
