---
title: "assigning ASVs with DADA2"
output: html_document
date: "r Sys.Date()"
editor_options: 
  chunk_output_type: console
---



```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE,
                      fig.align = "center",
                      fig.path = "../figures/01_DADA2")
```

#symbolically link our data files

We need to have access to our Fastq files for out project
```
#ran these commands to symbolically link

cd SalinityGradient_16S
mkdir data
cd data
mkdir 01_DADA2
cd 01_DADA2
mkdir 01_raw_gzipped_fastqs

# Now, let's run a for loop to symbolically link the data! 
for FILE in `ls /workdir/in_class_data/raw_gzipped_seqs/*.fastq.gz`
  do
  ln -s $FILE /workdir/<your_netID>/SalinityGradient_16S/data/01_DADA2/01_raw_gzipped_fastqs/
  done
```

# load R libraries
```{r load libraries}
#efficiently load with pac man
pacman::p_load(tidyverse, dada2, phyloseq, DT, devtools, install=FALSE)
```

# Load in the raw sequencing data
```{r}
#set the path

```


# Visualize and inspect sequencing data to assess the quality of the sequencing reads.

# Filter by the quality and trim primers from sequencing data

# Write new fastq files that include high quality and trimmed sequences 

# Re-inspect the quality of the reads (intuition check to make sure that we have it)

6. Infer the error rate on the forward R1 and reverse R2 reads.
**DADA2 building the error model**
7. Apply the error model and infer ASVs on the forward and reverse reads individually.
8. Merge forward and reverse ASVs into "contiguous ASVs".
9. Generate an ASV count table ('otu_table' for phyloseq.)


output that we need:
1. ASV count table: 'otu_table
